src/borg/repository.py

```diff
@@ -138,6 +139,7 @@ def __enter__(self):
         if self.do_create:
             self.do_create = False
             self.create(self.path)
+            self.created = True
```

Comment:
You can reopen a repository, but you can't create it multiple times. This is used in the tests in quite a few spots.